src/engagement_db_coda_sync/configuration.py

```diff
@@ -1,37 +1,112 @@
-from dataclasses import dataclass
-from typing import Callable, Optional
-
-from core_data_modules.data_models import CodeScheme
+class CodeSchemeConfiguration:
+    def __init__(self, code_scheme, auto_coder=None, coda_code_schemes_count=1):
+        """
+        Configures one normal code scheme in a Coda dataset.
 
+        There is no need to create this type of configuration to set WS - Correct Dataset code schemes.
+        Use the global configuration option in `CodaSyncConfiguration` for this instead.
 
-@dataclass
-class CodeSchemeConfiguration:
-    code_scheme: CodeScheme
-    auto_coder: Optional[Callable[[str], str]]
-    coda_code_schemes_count: Optional[int] = 1
+        :param code_scheme: A Code scheme that should appear in Coda and which messages can be labelled under.
+        :type code_scheme: core_data_modules.data_models.CodeScheme
+        :param auto_coder: A function which, given a message text, returns a suggestion for the best code to assign
+                           to this message. If this is provided, then when adding new, unlabelled messages to Coda:
+                            - This auto_coder will be called with the message text to get a suggested code.
+                            - The Code which has this suggestion as a match_value will be retrieved from the
+                              `code_scheme`.
+                            - A new Label will be created from this Code, and assigned to the message in Coda and in
+                              the engagement database. This label will be added in the 'unchecked' state, meaning it
+                              will still require human review before it should be used in analysis.
+                           If None, messages will be added without creating any new, suggested Labels.
+                           If a message already has labels (checked or unchecked), these existing labels will be added
+                           to Coda instead.
+        :type auto_coder: (function of str -> str) | None
+        :param coda_code_schemes_count: The number of copies of this code scheme that should appear in Coda.
+                                        (Multiple copies allow multiple different labels to be assigned under this
+                                         code scheme).
```

Comment:
I don't think we need to specify the default here because it is defined in the function signature. If we write it here too then there's a risk of confusion if we ever change the default in the function sig. but forget to change it here.